esi/service_description.py

```diff
@@ -0,0 +1,189 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from openstack import service_description
+import warnings
+
+import os_service_types
+
+from openstack import _log
+from openstack import exceptions
+from openstack import proxy as proxy_mod
+from openstack import warnings as os_warnings
+
+class ESIServiceDescription(service_description.ServiceDescription):
+
+    def _make_proxy(self, instance):
```

Comment:
I created a subclass ESIServiceDescription to override the _make_proxy(). See the code that has been commented out. config.has_service() method checks if the service is officially listed in openstack. If not, it will disable the service (our 'lease' service).

Reply:
Since this SDK is explicitly only designed to work with the ESI service, does it even need to know about other services? If not, it feels like maybe this code could be simplified to explicitly look for the 'lease' service.

Reply:
Makes sense!

Reply:
I've removed service_description.py and made changes to lease_service.py to override _make_proxy()